src/Agent.Worker/DiagnosticLogManager.cs

```diff
@@ -105,6 +105,14 @@ public sealed class DiagnosticLogManager : AgentService, IDiagnosticLogManager
                 File.Copy(agentLogFile, destination);
             }
 
+            // Copy cloud-init log files from linux machines
+            if (PlatformUtil.RunningOnLinux)
+            {
+                executionContext.Debug("Dumping cloud-init logs.");
+                await DumpCloudInitLogs(jobStartTimeUtc, HostContext.GetDirectory(WellKnownDirectory.Diag));
```

Comment:
Please add fallback - so if there is an error, it won't crush agent, but will print an error message in logs.